src/transformers/models/whisper/generation_whisper.py

```diff
@@ -884,39 +905,82 @@ def _prepare_segments(prompt_ids, batch_size, generation_config):
 
         return current_segments
 
-    def _postprocess_outputs(self, seek_outputs, decoder_input_ids, return_token_timestamps, generation_config):
+    def _postprocess_outputs(
+        self, seek_outputs, decoder_input_ids, return_token_timestamps, generation_config, is_shortform
+    ):
         # remove all previously passed decoder input ids
         if isinstance(seek_outputs, torch.Tensor):
```

Comment:
Could we first set a `start_idx` as follows:
```python
start_idx = decoder_input_ids.shape[-1] if not is_shortform else torch.tensor(0)
```

And then remove all the `if/else` statements on `is_shortform` in the rest of the method?

E.g. the following:
```python
            if not is_shortform:
                seek_outputs = seek_outputs[:, decoder_input_ids.shape[-1] :]
                return seek_outputs, seek_outputs
            else:
                return seek_outputs, seek_outputs
```

Becomes:
```python
            seek_outputs = seek_outputs[:, start_idx :]
            return seek_outputs, seek_outputs
```

Same for the token timestamps and sequences later in the method